moebot_bot/bot/bot.go

```diff
@@ -148,6 +149,63 @@ func messageCreate(session *discordgo.Session, message *discordgo.MessageCreate)
 	//distributeTickets(guild, message, session, messageTime)
 }
 
+func messageReactionAdd(session *discordgo.Session, reactionAdd *discordgo.MessageReactionAdd) {
+	polls, err := db.PollsOpenQuery()
+	if err != nil {
+		return
+	}
+	for _, p := range polls {
+		if p.MessageId == reactionAdd.MessageID {
+			p.Options, err = db.PollOptionQuery(p.Id)
+			if err != nil {
+				log.Println("Cannot retrieve poll options informations", err)
+				return
+			}
+			//If the user is reacting to a poll, we check if he has already cast a vote and remove it
+			handleSingleVote(session, p, reactionAdd)
+			return
+		}
+	}
+}
+
+func handleSingleVote(session *discordgo.Session, poll *db.Poll, reactionAdd *discordgo.MessageReactionAdd) {
+	message, err := session.ChannelMessage(poll.ChannelId, poll.MessageId)
+	if err != nil {
+		log.Println("Cannot retrieve poll message informations", err)
+		return
+	}
+	if message.Author.ID == reactionAdd.UserID {
+		return //The bot is modifying its own reactions
+	}
+	for _, r := range message.Reactions {
+		if !reactionIsOption(poll.Options, r.Emoji.Name) {
+			continue
+		}
+		//Getting a list of users for every reaction
+		users, err := session.MessageReactions(poll.ChannelId, poll.MessageId, r.Emoji.Name, 100)
```

Comment:
As with a lot of discord things we're going to have to find a way to make this work when stuff gets above 100. Not an issue, just a note for the future.